docs/DeveloperGuide.md

```diff
@@ -2,14 +2,266 @@
+
+Step 3:
+The `execute()` function of `AddCommand` will run, calling `addNewFlashcard(questionText, answerText)`
+of class `FlashcardList` to create and add the new flashcard to the list.
+
+Then it will also call `printSuccessfulAddMessage(questionText, answerText)` of class `Ui`
+to display text indicating the successful adding function to the user.
+
+At this point, the adding process is completed and the program is ready to take another
+command.
+
+The following sequence diagram show how the add operation works:
+![AddFlashcard-0.png](umlDiagrams%2FAddFlashcard-0.png)
+
+#### Reason for current implementation
+
+Through using `AddCommand` class, which extends `Command` class it increases the level of
+abstraction as the code can now perform the various commands on a class level.
+
+Moreover, since the creating of new `Flashcard` of object and adding of the newly created
+flashcard are both done in the same class as where the flashcards are stored, this reduces
+coupling in the program as the `AddCommand` will not have access to the inner structure of
+`FlashcardList`, which stores the list of flashcards.
+
+#### Alternative implementation
+
+- Alternative 1: Have the add command function directly in `FlashcardList`
+    - Pros: Easy to implement
+    - Cons: Will require another function in another program to differentiate it from other
+      commands
+- Alternative 2: Have the constructor of `Flashcard` include adding the card to list of flashcards
+    - Pros: Simplifies code
+    - Cons: Will cause trouble when temporary flashcard (that need not be stored) are
+      created
+
+### Delete Flashcard Feature
+
+#### Current implementation
+
+The current delete flashcard allows the user to remove a flashcard from the list of flashcards,
+it is implemented through the following steps:
+
+Step 1: The input of user is collected by `getUserCommand()` inside class `Ui`.
+
+Step 2: The input string will be converted into a `Command` object by being passed through
+`parseCommand(String userInput)` inside `Parser`.
+
+In this case, a `DeleteCommnad` will be created and returned.
+
+Step 3: The `execute()` function of `DeleteCommand` will run, creating a copy of the list of flashcards.
+Then `findFlashcards(flashcards, query)` is called to find the flashcards with questions matching the query,
+before calling `printFlashcardList(matchingFlashcards)` to display the found flashcards.
+
+User input is taken to get the index of the flashcard to be removed. `deleteFlashcard` is called from
+class `flashcardList` to remove the flashcard from the original list of flashcards. Finally
+`printSuccessfulDelete` is called from class `Ui` to indicate a successful removal of the flashcard.
+
+The deletion process is now completed and the program will await another command.
+
+An overview of how the Delete operation works is shown with the following sequence diagram
+![DeleteFlashcard.png](umlDiagrams%2FDeleteFlashcard.png)
```

Comment:
Perhaps you could extend the activation bar for "UI" and "Delete Command" all the way until the last call since you did not terminate it in between calls